tests/e2e/cucumber/publicLinkToFolderJourney.feature

```diff
@@ -0,0 +1,25 @@
+Feature: Create public link to folder
+  As a user
+  I want to share folder resources through a publicly accessible link with role
+  So that unauthorized user can access the public link based on role set
+
+  Background:
+    Given the following users have been created
+      | Alice |
+
+    Scenario: Alice creates public link to a folder
+      When "Alice" logs in
+      And "Alice" opens the "files" app
+      And "Alice" creates the following folder
+        | folderPublic |
+      And "Alice" uploads the following resource
+        | resource  | to           |
+        | lorem.txt | folderPublic |
+      Then "Alice" should see the following resource
+        | folderPublic/lorem.txt |
+      When "Alice" creates public link to the following resource via the sidebar panel
+        | resource     | name         | role   | dateOfExpiration | password |
+        | folderPublic | myPublicLink | editor | 2022-02-19       | 12345    |
```

Comment:
The test step needs to be able to specify a relative date, like "tomorrow" "+2 days" etc.
The test code can work out what actual date that is at run-time, and select that date in the UI.
Otherwise the test will break next week.

Reply:
yes @phil-davis ji i am working on for both.

Reply:
is a good idea, but sometimes there are failures due to possible time differences. I'm not sure, but it has to be taken into account.

Reply:
yes, the test needs to run OK on any day of the week, day of the month, and month of the year. We want the test to choose a date in the future, to be valid. If the CI runs on 31 December then the test code has to be able to select the next year, the first month of the next year, and one of the days in that month. Similar if the test runs near the end of a month, it has to select a date in the next month.

These UI tests with dates are difficult/tricky to code. If the code actually navigates and clicks all the various elements in the date-picker UI then it has to be quite flexible to "do the right thing" for each possible week/month/year boundary.